### PR TITLE
Rename of structs and methods

### PR DESCRIPTION
**Pull request by krishnasrinivas**
Should be merged after https://github.com/minio/minio/pull/5959

## Description

In the recent PR https://github.com/minio/minio/pull/5959 StorageAPI details were taken out of erasure-* files i.e erasure related code deals only with bitrotWriter and bitrotReader. (In future bitrotWriter will be io.Writer compatible and bitrotReader will be io.Reader compatible so erasure related files can separated into its own package.)

Hence the following struct

```
type ErasureStorage struct {
        erasure                  reedsolomon.Encoder
        dataBlocks, parityBlocks int
        blockSize                int64
}
```

is renamed to just `Erasure`
```
type Erasure struct {
        encoder                  reedsolomon.Encoder
        dataBlocks, parityBlocks int
        blockSize                int64
}
```

Its field `erasure` renamed to `encoder`

Its methods also renamed accordingly:

ErasureStorage.CreateFile -> Erasure.Encode
ErasureStorage.ReadFile -> Erasure.Decode
ErasureStorage.HealFile -> Erasure.Heal (could not find a better name)

ErasureStorage.ErasureEncode -> Erasure.EncodeData
ErasureStorage.ErasureDecodeDataBlocks -> Erasure.DecodeDataBlocks
ErasureStorage.ErasureDecodeDataAndParityBlocks -> Erasure.DecodeDataAndParityBlocks

files renamed accordingly:
erasure-createfile.go -> erasure-encode.go
erasure-readfile.go -> erasure-decode.go
erasure-healfile.go -> erasure-heal.go (could not find a better name)

-----------------------------------------------------

Renamed BackendType:
FS -> BackendFS
Erasure -> BackendErasure

## Types of changes

- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)

## Checklist:

- [ ] My code follows the code style of this project.
- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.
- [ ] I have added unit tests to cover my changes.
- [ ] I have added/updated functional tests in [mint](https://github.com/minio/mint). (If yes, add `mint` PR # here: )
- [ ] All new and existing tests passed.

### CODE REVIEW

**Comment by minio-ops** [comment]
## Mint Automation
| Test | Result |
|:--- |:--- |
| mint-gateway-s3.sh | :heavy_check_mark: |
| mint-xl.sh | :heavy_check_mark: |
| mint-gateway-nas.sh | :heavy_check_mark: |
| mint-large-bucket.sh | :heavy_check_mark: |
| mint-worm.sh | :heavy_check_mark: |
| mint-fs.sh | :heavy_check_mark: |
| mint-dist-xl.sh | :heavy_check_mark: |
| mint-gateway-azure.sh | :x: [more...](#fc54e79e-1c70-4a2c-97c1-8ad964bb43a3) |
<a name="fc54e79e-1c70-4a2c-97c1-8ad964bb43a3"></a>
### 6230-c524f61/mint-gateway-azure.sh.log:
```
Running with
SERVER_ENDPOINT:      minikube:30945
ACCESS_KEY:           minioazure
SECRET_KEY:           ***REDACTED***
ENABLE_HTTPS:         0
SERVER_REGION:        us-east-1
MINT_DATA_DIR:        /mint/data
MINT_MODE:            full
ENABLE_VIRTUAL_STYLE: 0

To get logs, run 'docker cp 83e9d7665bad:/mint/log /tmp/mint-logs'
(1/13) Running awscli tests ... done in 3 minutes and 0 seconds
(2/13) Running aws-sdk-go tests ... done in 2 seconds
(3/13) Running aws-sdk-java tests ... done in 2 seconds
(4/13) Running aws-sdk-php tests ... done in 5 minutes and 0 seconds
(5/13) Running aws-sdk-ruby tests ... done in 38 seconds
(6/13) Running mc tests ... done in 7 minutes and 7 seconds
(7/13) Running minio-dotnet tests ... done in 5 minutes and 48 seconds
(8/13) Running minio-go tests ... FAILED in 8 minutes and 7 seconds
{
  "args": {
    "policy": "{\"expiration\":\"2018-08-12T01:10:39.775Z\",\"conditions\":[[\"eq\",\"$bucket\",\"minio-go-test-3gqcro06jisy2ecf\"],[\"eq\",\"$key\",\"ymihgei3bwpjack2eiymgnl9cz60bo\"],[\"eq\",\"$Content-Type\",\"binary/octet-stream\"],[\"eq\",\"$x-amz-meta-9iy0voei6h9t5l6bxl3mgexel5gl6j\",\"rw5fek33rjzlxr1lt1mteooggesozc\"],[\"content-length-range\", 10, 1048576]]}"
  },
  "duration": 414,
  "error": "400 Bad Request",
  "function": "PresignedPostPolicy(policy)",
  "message": "Http request failed",
  "name": "minio-go: testPresignedPostPolicy",
  "status": "FAIL"
}

Executed 7 out of 13 tests successfully.
```

**Comment by krishnasrinivas** [comment]
Some more renames needs to be done inside the test.go files

File: cmd/bitrot.go [hunk 1]
```diff
@@ -0,0 +1,192 @@
...
+	case HighwayHash256:
+		hh, _ := highwayhash.New(magicHighwayHash256Key) // New will never return error since key is 256 bit
+		return hh
+	default:
+		logger.CriticalIf(context.Background(), errors.New("Unsupported bitrot algorithm"))
+		return nil
+	}
+}
+
+// Available reports whether the given algorihm is available.
```
Comment (poornas) [inline cmd/bitrot.go]: typo -> algorihm

File: cmd/storage-rpc_test.go [hunk 2]
```diff
@@ -347,13 +348,14 @@ func testStorageAPIReadFile(t *testing.T, storage StorageAPI) {
...
 	for i, testCase := range testCases {
-		result := make([]byte, 100)
-		n, err := storage.ReadFile(testCase.volumeName, testCase.objectName, testCase.offset, result, nil)
+		result = result[testCase.offset:3]
+		_, err := storage.ReadFile(testCase.volumeName, testCase.objectName, testCase.offset, result, nil)
 		expectErr := (err != nil)
-		result = result[:n]
 
 		if expectErr != testCase.expectErr {
+			fmt.Println(err)
```
Comment (poornas) [inline cmd/storage-rpc_test.go]: remove this?

**Review by poornas — CHANGES_REQUESTED** [review]

**Comment by donatello** [comment]
> Some more renames needs to be done inside the test.go files

@krishnasrinivas Is this presently ready for review?

**Comment by krishnasrinivas** [comment]
> @krishnasrinivas Is this presently ready for review?

@donatello yes

**Review by donatello — APPROVED** [review]

**Review by poornas — APPROVED** [review]

**Pull request merged**